### ISSUE TITLE
MySQL syntax error with some LEFT JOINs

### ISSUE THREAD

**Issue by RobertLowe** [metabase/metabase#9632]
- Your browser and the version: N/A
- Your operating system: N/a
- Your databases: MySQL
- Metabase version: 0.31
- Metabase hosting environment: Heroku
- Metabase internal database: Postgres

With our dataset I've gotten MySQL syntax errors from the question builder, here's an example:

```
SELECT `users`.`profile.schoolId` AS `profile.schoolId`, 
        sum(`schools__via__profile.schoolId`.`cost`) AS `sum`
FROM `users` 
LEFT JOIN `schools` `schools__via__profile`.`schoolId` ON `users`.`profile.schoolId` = `schools__via__profile.schoolId`.`_id` 
WHERE `users`.`profile.schoolName` IS NOT NULL 
GROUP BY `users`.`profile.schoolId` 
ORDER BY `users`.`profile.schoolId` ASC
```

Where as I would of expected it to generate something like:

```
SELECT `users`.`profile.schoolId` AS `profile.schoolId`,
        sum(`schools__via__profile`.`cost`) AS `sum`
FROM `users`
LEFT JOIN `schools` AS `schools__via__profile` ON `users`.`profile.schoolId` = `schools__via__profile`.`_id`
WHERE `users`.`profile.schoolName` IS NOT NULL
GROUP BY `users`.`profile.schoolId`
ORDER BY `users`.`profile.schoolId` ASC;
```

It appears the LEFT JOIN is missing the AS alias and ='s right comparison naming incorrectly

**Comment by camsaul** [comment]
@RobertLowe can you screenshot the query you were trying to run from the Question Builder? For debugging purposes

**Comment by RobertLowe** [comment]
These were generated by the xray:

![Screen Shot 2019-03-29 at 6 25 06 PM](https://user-images.githubusercontent.com/601141/55265885-93150380-5250-11e9-81cf-3a98afc80923.png)
![Screen Shot 2019-03-29 at 6 26 23 PM](https://user-images.githubusercontent.com/601141/55265891-97412100-5250-11e9-89fc-21991c47e57c.png)

Here's a stacktrace:

```
[1mWARN metabase.query-processor
[0m :: 
[31m{:status :failed,
:class java.util.concurrent.ExecutionException,
:error
"com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: parse sql '-- Metabase:: userID: 1 queryType: MBQL queryHash: 593d2162f822fe7a146a3241abdf248bdbc73d0815ebc0d8156ab19baf4e34ec\nSELECT count(distinct `users__via__skipWebCam.authori`.`_id`) AS `count` FROM `reservations` LEFT JOIN `users` `users__via__skipWebCam`.`authori` ON `reservations`.`skipWebCam.authorizedBy` = `users__via__skipWebCam.authori`.`_id`' error: syntax error at position 253",
:stacktrace
("java.util.concurrent.FutureTask.report(FutureTask.java:122)"
"java.util.concurrent.FutureTask.get(FutureTask.java:192)"
"clojure.core$deref_future.invokeStatic(core.clj:2292)"
"clojure.core$future_call$reify__8097.deref(core.clj:6894)"
"clojure.core$deref.invokeStatic(core.clj:2312)"
"clojure.core$deref.invoke(core.clj:2298)"
"--> driver.generic_sql.query_processor$cancellable_run_query.invokeStatic(query_processor.clj:620)"
"driver.generic_sql.query_processor$cancellable_run_query.invoke(query_processor.clj:612)"
"driver.generic_sql.query_processor$run_query.invokeStatic(query_processor.clj:637)"
"driver.generic_sql.query_processor$run_query.invoke(query_processor.clj:632)"
"driver.generic_sql.query_processor$do_with_auto_commit_disabled.invokeStatic(query_processor.clj:675)"
"driver.generic_sql.query_processor$do_with_auto_commit_disabled.invoke(query_processor.clj:664)"
"driver.generic_sql.query_processor$do_in_transaction$fn__57914.invoke(query_processor.clj:680)"
"driver.generic_sql.query_processor$do_in_transaction.invokeStatic(query_processor.clj:679)"
"driver.generic_sql.query_processor$do_in_transaction.invoke(query_processor.clj:678)"
"driver.generic_sql.query_processor$run_query_without_timezone.invokeStatic(query_processor.clj:696)"
"driver.generic_sql.query_processor$run_query_without_timezone.invoke(query_processor.clj:695)"
"driver.generic_sql.query_processor$execute_query$fn__57938.invoke(query_processor.clj:723)"
"driver.generic_sql.query_processor$do_with_try_catch.invokeStatic(query_processor.clj:659)"
"driver.generic_sql.query_processor$do_with_try_catch.invoke(query_processor.clj:654)"
"driver.generic_sql.query_processor$execute_query.invokeStatic(query_processor.clj:720)"
"driver.generic_sql.query_processor$execute_query.invoke(query_processor.clj:716)"
"driver$fn__29943$G__29750__29950.invoke(driver.clj:106)"
"query_processor$execute_query.invokeStatic(query_processor.clj:66)"
"query_processor$execute_query.invoke(query_processor.clj:60)"
"query_processor.middleware.mbql_to_native$mbql__GT_native$fn__32477.invoke(mbql_to_native.clj:36)"
"query_processor.middleware.annotate$add_column_info$fn__31198.invoke(annotate.clj:300)"
"query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__31939.invoke(cumulative_aggregations.clj:57)"
"query_processor.middleware.limit$limit$fn__32430.invoke(limit.clj:17)"
"query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__37552.invoke(results_metadata.clj:87)"
"query_processor.middleware.format_rows$format_rows$fn__32420.invoke(format_rows.clj:26)"
"query_processor.middleware.add_dimension_projections$add_remapping$fn__30542.invoke(add_dimension_projections.clj:236)"
"query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__30830.invoke(add_row_count_and_status.clj:15)"
"query_processor.middleware.driver_specific$process_query_in_context$fn__32181.invoke(driver_specific.clj:12)"
"query_processor.middleware.resolve_driver$resolve_driver$fn__35350.invoke(resolve_driver.clj:15)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31376$fn__31377.invoke(bind_effective_timezone.clj:9)"
"util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
"util.date$call_with_effective_timezone.invoke(date.clj:77)"
"query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31376.invoke(bind_effective_timezone.clj:8)"
"query_processor.middleware.store$initialize_store$fn__37567$fn__37568.invoke(store.clj:11)"
"query_processor.store$do_with_new_store.invokeStatic(store.clj:34)"
"query_processor.store$do_with_new_store.invoke(store.clj:30)"
"query_processor.middleware.store$initialize_store$fn__37567.invoke(store.clj:10)"
"query_processor.middleware.cache$maybe_return_cached_results$fn__31798.invoke(cache.clj:127)"
"query_processor.middleware.catch_exceptions$catch_exceptions$fn__31846.invoke(catch_exceptions.clj:64)"
"query_processor$process_query.invokeStatic(query_processor.clj:213)"
"query_processor$process_query.invoke(query_processor.clj:209)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:322)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
"query_processor$fn__37732$process_query_and_save_execution_BANG___37737$fn__37738.invoke(query_processor.clj:354)"
"query_processor$fn__37732$process_query_and_save_execution_BANG___37737.invoke(query_processor.clj:340)"
"query_processor$fn__37756$process_query_and_save_with_max_BANG___37761$fn__37762.invoke(query_processor.clj:373)"
"query_processor$fn__37756$process_query_and_save_with_max_BANG___37761.invoke(query_processor.clj:369)"
"api.dataset$fn__44259$fn__44262.invoke(dataset.clj:45)"
"api.common$fn__23158$invoke_thunk_with_keepalive__23163$fn__23164$fn__23165.invoke(common.clj:435)"),
:query
{:type "query",
:query {:source-table 636, :aggregation [["distinct" ["fk->" 13296 13520]]]},
:parameters [],
:constraints {:max-results 10000, :max-results-bare-rows 2000},
:info
{:executed-by 1,
 :context :ad-hoc,
 :card-id nil,
 :nested? false,
 :query-hash [89, 61, 33, 98, -8, 34, -2, 122, 20, 106, 50, 65, -85, -33, 36, -117, -37, -57, 61, 8, 21, -21, -64, -40, 21, 106, -79, -101, -81, 78, 52, -20],
 :query-type "MBQL"}},
:preprocessed
{:type :query,
:database 6,
:query
{:source-table 636,
 :aggregation [[:distinct [:fk-> [:field-id 13296] [:field-id 13520]]]],
 :join-tables ({:join-alias "users__via__skipWebCam.authori", :table-id 617, :fk-field-id 13296, :pk-field-id 13520})},
:constraints {:max-results 10000, :max-results-bare-rows 2000},
:info
{:executed-by 1,
 :context :ad-hoc,
 :nested? false,
 :query-hash [89, 61, 33, 98, -8, 34, -2, 122, 20, 106, 50, 65, -85, -33, 36, -117, -37, -57, 61, 8, 21, -21, -64, -40, 21, 106, -79, -101, -81, 78, 52, -20],
 :query-type "MBQL"},
:driver {},
:settings {}},
:native
{:query
"SELECT count(distinct `users__via__skipWebCam.authori`.`_id`) AS `count` FROM `reservations` LEFT JOIN `users` `users__via__skipWebCam`.`authori` ON `reservations`.`skipWebCam.authorizedBy` = `users__via__skipWebCam.authori`.`_id`",
:params nil}}
[0m
```

**Comment by camsaul** [comment]
This should be fixed in 0.32.9

**Comment by mazameli** [comment]
@RobertLowe Have you upgraded your Metabase instance and seeing if this is fixed for you? It should have been since version 0.32.9 (latest is 0.33.4).

**Comment by camsaul** [comment]
Closing as fixed in 0.32.9
